docs/data/joy/components/autocomplete/autocomplete.md

```diff
@@ -0,0 +1,394 @@
+- `listbox`: a default [popup](/base/react-popper/) with its `role` set to `listbox`.
+  - `option`: an option with its `role` set to `option`.
+  - `loading`: an indicator that appears when `loading` is set to `true`.
+  - `noOptions`: a section that appears when there is no options to show.
+
+```html
+<div class="JoyAutocomplete-root">
+  <!-- The div below comes from the `renderInput` function -->
+  <div>
+    <!-- The span is the text input's start decorator -->
+    <span>
+      <span class="JoyAutocomplete-limitTag">…</span>
+    </span>
+    <input role="combobox" aria-autocomplete="list" />
+    <!-- The span is the text input's end decorator -->
+    <span>
+      <button tabindex="-1" class="JoyAutocomplete-clearIndicator">
+        <svg data-testid="CloseIcon">…</svg>
+      </button>
+      <button tabindex="-1" class="JoyAutocomplete-popupIndicator">
+        <svg data-testid="ArrowDropdownIcon">…</svg>
+      </button>
+    </span>
+  </div>
+</div>
+<!-- By default, the popup displays with portal -->
+<ul role="listbox" class="Autocomplete-listbox">
+  <li class="Autocomplete-loading">…</li>
+  <li class="Autocomplete-noOptions">…</li>
+  <li
+    role="option"
+    class="Autocomplete-option"
+    aria-selected="true"
+    aria-disabled="false"
+  >
+    …
+  </li>
+  …
+</ul>
+```
+
+### Overriding the root slot
+
+Use the `component` prop to override the `root` slot with a custom element. For example, the following code snippet replaces the default `<div>` with a `<fieldset>`:
+
+```js
+<Autocomplete component="fieldset" />;
+
+// renders as:
+<fieldset class="JoyAutocomplete-root">…</fieldset>;
+```
+
+### Overriding other slots
+
+Use the `components` with keys (the target slots) and values (the custom elements). For example, the following code snippet replaces the default `<ul>` and `<li>` with `div`:
+
+```js
+<Autocomplete
+  components={{
+    listbox: 'div',
+    option: 'div',
+  }}
+>
+
+// renders as:
+<div class="JoyAutocomplete-root">…</div>
+<div class="JoyAutocomplete-listbox">
+  <div class="JoyAutocomplete-option">…</div>
+  …
+</div>
+```
+
+## Customization
+
+### Options structure
+
+By default, the `options` accepts an array of `string` or `{ label: string }`:
+
+```js
+const options = [
+  { label: 'The Godfather', id: 1 },
+  { label: 'Pulp Fiction', id: 2 },
+];
+// or
+const options = ['The Godfather', 'Pulp Fiction'];
+```
+
+However, you can use different structures by providing a `getOptionLabel` prop:
+
+```js
+const options = [
+  { title: 'Pulp Fiction', id: 2 },
+  // ...
+];
+
+<Autocomplete getOptionLabel={option => option.title}>
+```
+
+### Option appearance
+
+To customize the appearance of the options, use `renderOption` prop in combination with the `AutocompleteOption` component as an option container.
+
+{{"demo": "CountrySelect.js"}}
+
+### Input appearance
+
+To customize the appearance of the input, use `renderInput` prop that returns an `Input` component.
```

Comment:
```suggestion
To customize the appearance of the input, use `renderInput` prop that returns an `Input` component⎯that way, you'll be able to use any global variant or the `sx prop` to further customize it.
```

I think we could do something like this, to extend a bit more on customizability. Given the `Input` component is there being used, I'm assuming I can just throw both the variants and the `sx` in there?

Reply:
👍 Great suggestion.